dart/lib/src/event_processor/enricher/io_enricher_event_processor.dart

```diff
@@ -37,17 +41,15 @@ class IoEnricherEventProcessor implements EnricherEventProcessor {
   }
 
   @override
-  SentryEvent? apply(SentryEvent event, Hint hint) {
-    // Amend app with current memory usage, as this is not available on native.
-    final app = _getApp(event.contexts.app);
-
-    final contexts = event.contexts.copyWith(
-      device: _getDevice(event.contexts.device),
-      operatingSystem: _getOperatingSystem(event.contexts.operatingSystem),
-      runtimes: _getRuntimes(event.contexts.runtimes),
-      app: app,
-      culture: _getSentryCulture(event.contexts.culture),
-    );
+  Future<SentryEvent?> apply(SentryEvent event, Hint hint) async {
+    final contexts = event.contexts;
+
+    contexts.device = await _getDevice(event.contexts.device);
+    contexts.operatingSystem =
+        _getOperatingSystem(event.contexts.operatingSystem);
+    contexts.runtimes = _getRuntimes(event.contexts.runtimes);
+    contexts.app = _getApp(event.contexts.app);
+    contexts.culture = _getSentryCulture(event.contexts.culture);
```

Comment:
I didnt know we can mutate directly

Reply:
The contexts are just a subclassed map. I found in the CPU profiler that copying here was taking some time, so this is an easy perf improvement we can do.